Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1137,6 +1137,95 @@ class BrowserViewController: UIViewController {
+        if(homePanelController == nil){
+            tab( (tabManager.selectedTab)!, didSelectFindInPageForSelection: "")
+        }
+    }
+    
+    func openLocation(){
+        urlBar.tabLocationViewDidTapLocation(urlBar.locationView)
+    }
+    
+    func newTab(){
+        openBlankNewTabAndFocus(isPrivate: false)
+    }
+    func newPrivateTab(){
+        openBlankNewTabAndFocus(isPrivate: true)
+    }
+    
+    func closeTab(){
+        if(tabManager.tabs.count > 1){
+            tabManager.removeTab(tabManager.selectedTab!);
+        }
+        else{
+            //need to close the last tab and show the favorites screen thing
+        }
+    }
+    
+    func nextTab(){
+        if(tabManager.selectedIndex < (tabManager.tabs.count - 1) ){
+            tabManager.selectTab(tabManager.tabs[tabManager.selectedIndex+1])
+        }
+        else{
+            if(tabManager.tabs.count > 1){
+                tabManager.selectTab(tabManager.tabs[0]);
+            }
+        }
+    }
+    
+    func previousTab(){
+        if(tabManager.selectedIndex > 0){
+            tabManager.selectTab(tabManager.tabs[tabManager.selectedIndex-1])
+        }
+        else{
+            if(tabManager.tabs.count > 1){
+                tabManager.selectTab(tabManager.tabs[tabManager.count-1])
+            }
+        }
+    }
+
+    override var keyCommands: [UIKeyCommand]? {
+        if #available(iOS 9.0, *) {
+            return [
+                UIKeyCommand(input: "r", modifierFlags: .Command, action: #selector(BrowserViewController.reloadTab), discoverabilityTitle: Strings.ReloadPageTitle),
+                UIKeyCommand(input: "[", modifierFlags: .Command, action: #selector(BrowserViewController.goBack), discoverabilityTitle: Strings.BackTitle),
+                UIKeyCommand(input: "f", modifierFlags: .Command, action: #selector(BrowserViewController.findOnPage), discoverabilityTitle: Strings.FindTitle),
+                UIKeyCommand(input: "l", modifierFlags: .Command, action: #selector(BrowserViewController.openLocation), discoverabilityTitle: Strings.OpenLocationTitle),
+                UIKeyCommand(input: "t", modifierFlags: .Command, action: #selector(BrowserViewController.newTab), discoverabilityTitle: Strings.NewTabTitle),
+                UIKeyCommand(input: "p", modifierFlags: [.Command, .Shift], action: #selector(BrowserViewController.newPrivateTab), discoverabilityTitle: Strings.NewPrivateTabTitle),
+                UIKeyCommand(input: "w", modifierFlags: .Command, action: #selector(BrowserViewController.closeTab), discoverabilityTitle: Strings.CloseTabTitle),
+                UIKeyCommand(input: "\t", modifierFlags: .Control, action: #selector(BrowserViewController.nextTab), discoverabilityTitle: Strings.ShowNextTabTitle),
+                UIKeyCommand(input: "\t", modifierFlags: [.Control, .Shift], action: #selector(BrowserViewController.previousTab), discoverabilityTitle: Strings.ShowPreviousTabTitle),
+            ]
+        } else {
```

Comment:
I am going to suggest that we drop iOS8 support. We will drop it globally from the app soon. Don't invest in it.